packages/client/lib/execution/vmexecution.ts

```diff
@@ -268,12 +268,25 @@ export class VMExecution extends Execution {
               if (!this.started) {
                 throw Error('Execution stopped')
               }
+              const beforeTS = Date.now()
               const result = await this.vm.runBlock({
                 block,
                 root: parentState,
                 skipBlockValidation,
                 skipHeaderValidation: true,
               })
+              const afterTS = Date.now()
```

Comment:
I think this was discussed a few weeks back, but this is a good reason to start working on these worker processes. It would be great if we can spawn a new process / worker which only does the VM execution. Since the client also does other things (devp2p, rpc, etc.) this will sometimes block the VM (at least, I expect this to be) to execute another Promise, before jumping back into the VM again. So if the client is under heavy load (lots of peers, or lots of RPC requests) then this might give "false" warnings that the block execution is slow; it could be if you isolate this block it runs just fine in a single process (e.g. just run it in VM)

Reply:
Totally not writing this as an "against" argument, but just to note as my observation that the block execution times are actually pretty stable and repeatable (I also haven't noticed a (big) difference yet comparing runs with peers and re-execution with no peers (have switched pretty often recently), but also haven't paid too much explicit attention on this).